### ISSUE TITLE
Conversation separator shows up for two messages close together

### ISSUE THREAD

**Issue by gabedwrds** [ricochet-im/ricochet#161]
With 1.1.0 (OS X experimental package), the separator from #154 is sometimes displayed when two messages are separated by a very short time instead of >1 hour.

Timeline:
Some time ago: chat window opened
18:20: a conversation begins, separator is shown
18:53: separator shown inappropriately
18:57: separator shown inappropriately again

![screen shot 2015-04-10 at 18 58 17](https://cloud.githubusercontent.com/assets/1580619/7098796/feff8480-dfb3-11e4-9211-5695a52741ae.png)

**Comment by special** [comment]
This seems to happen rarely, in rapid back-and-forth conversation, and I think it has always appeared above an outgoing message. Maybe there is something that confuses the date comparison; the reordering logic, perhaps?

**Comment by gabedwrds** [comment]
Sometimes the calculated timespan is negative. I added some debug code to my client watching for this and just caught a -4.

I don't know exactly what causes this, but maybe the implementation should use something other than -1 to mean no previous message?

**Comment by special** [comment]
@gabedwrds good catch. This is happening because of the [message reordering](https://github.com/ricochet-im/ricochet/blob/master/src/core/ConversationModel.cpp#L173). The incoming message's timestamp is be the time when it was received, which will be a newer timestamp than the unacknowledged outgoing messages that show up after it.

I was going to suggest that we mess with the timestamp on reordered messages to enforce `prev_msg <= this_msg <= next_msg`, but that would ruin delta timestamps (on messages that the peer re-sent) in some cases.

I like either of your solutions. Note that there _is_ a case where the timespan can be < -3600:
 * Alice writes a message to Bob while he's offline
 * A few hours later, Bob starts his client and writes a message to Alice before they connect
 * On Alice's side, Bob's message is placed before hers, but the timestamp will be several hours later

But that might actually be a bug/flaw in the reordering: maybe we should take the delta timestamp into account so that Alice and Bob will get the same transcript in this case. Hmm.

**Comment by special** [comment]
Actually, we're already more relaxed for inbound authentication:

AuthHiddenServiceChannel.cpp:285
```
    } else if (publicKeyData.size() > 180) {
```

So let's just make the outbound test the same as that one.
